test/Market.t.sol

```diff
@@ -227,13 +225,14 @@ contract Market_replace is MarketReplaceTestHelper {
     });
     uint256 amount = _market.calculateCheckoutTotal(_amountToReplace);
     vm.startPrank(_namedAccounts.admin);
-    _removal.release(_removalIds[0], _originalCertificateAmount);
-    assertEq(_certificate.getNrtDeficit(), _originalCertificateAmount);
     _bpNori.deposit(_namedAccounts.admin, abi.encode(amount));
     _bpNori.approve(address(_market), amount);
+    _removal.release(_removalIds[0], _originalCertificateAmount);
+    vm.stopPrank();
   }
 
   function test() external {
+    assertEq(_certificate.getNrtDeficit(), _originalCertificateAmount);
```

Comment:
I moved this from the `setUp` function because assertions in the `setUp` use generic error messages if they fail, whereas within a test function you get the context of the failure (e.g., expected true to be false, etc).